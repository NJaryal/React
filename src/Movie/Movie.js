import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import MovieDetailed from 'src/MovieDetailed/MovieDetailed';
import { requestSingleMovieData } from 'src/MovieDetailed/MovieDetailedActions';
import { loadtoDos } from '../lib/service';

export class Movie extends Component {
  componentDidMount() {
    const { match } = this.props;
    const id = match && match.params ? match.params.id : 0;
    this.props.requestSingleMovieData(id);
    loadtoDos()
      .then(({ data }) => this.setState({ todos: data }))
      .catch(() => this.setState({ error: true }))
  }

  render() {
    const { singleMovieData } = this.props;
    return <MovieDetailed singleMovieData={singleMovieData} className="MovieComponent" data-test="Movie"/>;
  }
}

Movie.propTypes = {
  singleMovieData: PropTypes.object,
  match: PropTypes.object,
  requestSingleMovieData: PropTypes.func,
};

const mapStateToProps = state => ({
  singleMovieData: state.MovieDetailedReducer.singleMovieData,
});

const mapDispatchToProps = dispatch => ({
  requestSingleMovieData: (movieId) => {
    dispatch(requestSingleMovieData(movieId));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Movie);
