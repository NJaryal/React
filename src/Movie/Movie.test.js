import React from 'react';
import { shallow } from 'enzyme';
import { Movie } from './Movie';
import { checkProps } from '../../Utils';

const setup = () => {
  const component = shallow(<Movie />);
  return component;
};

describe('Movie component', () => {
  describe('Check PropTypes', () => {
    it('should not throw a warning', () => {
      const expectedprops = {
        match: {},
      };
      const propdError = checkProps(Movie, expectedprops);
      expect(propdError).toBeDefined();
    });
  });


  describe('Check PropTypes', () => {
    let component;
    beforeEach(() => {
      component = setup();
    });

    it('should render Component', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
