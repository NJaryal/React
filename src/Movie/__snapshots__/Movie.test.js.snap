// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component Check PropTypes should render Component 1`] = `undefined`;

exports[`Movie component should render Component 1`] = `undefined`;
